src/test/java/teammates/test/driver/HtmlHelper.java

```diff
@@ -287,4 +303,141 @@ private static boolean isVoidElement(String elementName){
                 || elementName.equals("meta");
     }
 
+    /**
+     * Injects values specified in configuration files to the appropriate placeholders.
+     */
+    public static String injectTestProperties(String content) {
+        return content.replace("${studentmotd.url}", Config.STUDENT_MOTD_URL)
+                      .replace("${version}", TP.TEAMMATES_VERSION)
+                      .replace("${test.admin}", TP.TEST_ADMIN_ACCOUNT)
+                      .replace("${test.student1}", TP.TEST_STUDENT1_ACCOUNT)
+                      .replace("${test.student2}", TP.TEST_STUDENT2_ACCOUNT)
+                      .replace("${test.instructor}", TP.TEST_INSTRUCTOR_ACCOUNT);
+    }
+
+    /**
+     * Processes the string from web page source for HTML comparison.
+     */
+    public static String processPageSourceForHtmlComparison(String content) {
+        return replaceUnpredictableValuesWithPlaceholders(
+                      suppressVariationsInInjectedValues(content));
+    }
+
+    /**
+     * Processes the string from web page source for regeneration of expected HTML.
+     */
+    public static String processPageSourceForExpectedHtmlRegeneration(String content, boolean isPart) {
+        return convertToStandardHtml(replaceInjectedValuesWithPlaceholders(content), isPart);
+    }
+
+    private static String suppressVariationsInInjectedValues(String content) {
+        return content // replace dev server admin absolute URLs (/_ah/...) with their relative counterparts
+                      .replace("\"" + TP.TEAMMATES_URL + "/_ah", "\"/_ah")
+                      // replace truncated long accounts with their original counterparts
+                      .replace(StringHelper.truncateLongId(TP.TEST_STUDENT1_ACCOUNT), TP.TEST_STUDENT1_ACCOUNT)
+                      .replace(StringHelper.truncateLongId(TP.TEST_STUDENT2_ACCOUNT), TP.TEST_STUDENT2_ACCOUNT)
+                      .replace(StringHelper.truncateLongId(TP.TEST_INSTRUCTOR_ACCOUNT), TP.TEST_INSTRUCTOR_ACCOUNT)
+                      .replace(StringHelper.truncateLongId(TP.TEST_ADMIN_ACCOUNT), TP.TEST_ADMIN_ACCOUNT);
+    }
+
+    private static String replaceUnpredictableValuesWithPlaceholders(String content) {
```

Comment:
A little explanation as to what this function does would be helpful. Eg: "Substitutes values that are different across various test runs with placeholders. These values are identified using their known format"